mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -176,10 +189,25 @@ pub enum SwapMsg {
     TakerPayment(Vec<u8>),
 }
 
+// Extension to SwapMsg with version added to negotiation exchange
+#[derive(Clone, Debug, Eq, Deserialize, PartialEq, Serialize)]
+pub enum SwapMsgExt {
+    NegotiationVersioned(NegotiationDataMsgVersion),
+    NegotiationReplyVersioned(NegotiationDataMsgVersion),
+}
```

Comment:
why not merge the new msgs into the old struct (and have a comment separating this is version x)?

Reply:
I did this on purpose: the old struct should be intact for old nodes to be able to parse it.
(I tried to test this though, in principle it could be tweaked to allow the merged struct to be parsed but I chose not to do this cause it involves knowledge about internal lib behaviour)